features/events/syslog/src/main/java/org/opennms/netmgt/syslogd/SyslogReceiverCamelNettyImpl.java

```diff
@@ -123,38 +128,63 @@ public void stop() throws InterruptedException {
         m_executor.shutdown();
     }
 
+    
+    
+    //Getter and setter for syslog handler
+    public SyslogConnectionHandler getSyslogConnectionHandlers() {
+	return m_syslogConnectionHandlers.get(0);
+    }
+
+    public void setSyslogConnectionHandlers(SyslogConnectionHandler handler) {
+	m_syslogConnectionHandlers = Collections.singletonList(handler);
+    }
+    
     /**
      * The execution context.
      */
     @Override
     public void run() {
-
         // Get a log instance
         Logging.putPrefix(Syslogd.LOG4J_CATEGORY);
 
         SimpleRegistry registry = new SimpleRegistry();
-        m_camel = new DefaultCamelContext(registry);
+
+	        //Adding netty component to camel inorder to resolve OSGi loading issues
+		NettyComponent nettyComponent=new NettyComponent();
+		m_camel = new DefaultCamelContext(registry);
+		m_camel.addComponent("netty", nettyComponent);
+
         try {
             m_camel.addRoutes(new RouteBuilder() {
                 @Override
                 public void configure() throws Exception {
-                    String from = String.format("netty:udp://%s:%s?sync=false&allowDefaultCodec=false&receiveBufferSize=%d&connectTimeout=%d",
+                    String from = String.format("netty:udp://%s:%s?sync=false&textline=true&receiveBufferSize=%d&connectTimeout=%d",
                         InetAddressUtils.str(m_host),
                         m_port,
                         Integer.MAX_VALUE,
                         SOCKET_TIMEOUT
                     );
-
+                    
                     from(from)
                     //.convertBodyTo(java.nio.ByteBuffer.class)
                     .process(new Processor() {
                         @Override
                         public void process(Exchange exchange) throws Exception {
-                            ChannelBuffer buffer = exchange.getIn().getBody(ChannelBuffer.class);
+                            ChannelBuffer buffer = ChannelBuffers.copiedBuffer(exchange.getIn().getBody().toString().getBytes());
```

Comment:
Is there a reason that this change is necessary? I think that you can keep the original line since the message body should already be a ChannelBuffer.